src/app/Shared/Services/CommandChannel.service.tsx

```diff
@@ -40,12 +40,21 @@ export class CommandChannel {
           if ((jsonResp.dashboardAvailable == true) && (jsonResp.datasourceAvailable == true)) {
             return forkJoin([getDatasourceURL, getDashboardURL])
           } else {
-              return throwError(new Error('ULRs unavailable'));
+            var errMessage = "";
+            if (jsonResp.dashboardAvailable == false) {
+              errMessage = "dashboard URL "
+            }
+            if (jsonResp.datasourceAvailable == false) {
+              errMessage = errMessage.concat(
+                (errMessage == "")? "datasource URL " : ", datasource URL ")
+            }
+            return throwError(errMessage.concat('missing'));
```

Comment:
This line uses a single-quote-delimited string, but above it's double-quotes. In TypeScript I tend to use single-quotes, other than when using backticks for string interpolation. Double quotes are used when the quoted text literal contains single-quotes/apostrophes, or in React TSX template fragments. Just a style thing.